### PR TITLE
Fix sentinel datasource init error in jacocoagent environment

### LINKED ISSUE #2476: sentinel datasource init error in jacocoagent environment
**Which Component**
Sentinel

**Describe the bug**
sentinel datasource init error in jacocoagent environment

**To Reproduce**
1. import spring-cloud-alibaba-sentinel module
2. config in application.yml like:
```
spring.cloud.sentinel.datasource.ds1.apollo.namespace-name=application
spring.cloud.sentinel.datasource.ds1.apollo.flow-rules-key=test-flow-rules
spring.cloud.sentinel.datasource.ds1.apollo.default-flow-rule-value=[]
spring.cloud.sentinel.datasource.ds1.apollo.rule-type=flow
```
3. run project with jacocoagent.jar, VM options like:
`-javaagent:D:\Users\Test\Desktop\jacoco-0.8.7\lib/jacocoagent.jar=includes=*,output=tcpserver,port=6271,address=127.0.0.1,append=true`
4. See error log
`2022-03-23 17:28:31.241 ERROR --- [ main] o.s.c.a.s.c.SentinelDataSourceHandler : [Sentinel Starter] DataSource ds1 multi datasource active and won't loaded: [apollo, $jacocoData]`
6. when i add isSynthetic check in DataSourcePropertiesConfiguration, see error log
`2022-03-23 17:43:20.571 ERROR --- [ main] o.s.c.a.s.c.SentinelDataSourceHandler : [Sentinel Starter] DataSource ds1 build error: Error creating bean with name 'ds1-sentinel-apollo-datasource': Error setting property values; nested exception is org.springframework.beans.NotWritablePropertyException: Invalid property '$jacocoData' of bean class [org.springframework.cloud.alibaba.sentinel.datasource.factorybean.ApolloDataSourceFactoryBean]: Bean property '$jacocoData' is not writable or has an invalid setter method. Does the parameter type of the setter match the return type of the getter?`

**Expected behavior**
sentinel datasource init success in jacocoagent environment

I create a pr to fix this problem, and wrote some unit test for DataSourcePropertiesConfiguration and SentinelDataSourceHandler

### PR DESCRIPTION
**Pull request by hnyyghk**
### Describe what this PR does / why we need it
Fix sentinel datasource init error in jacocoagent environment

### Does this pull request fix one issue?
Fixes #2476

### Describe how you did it
1. add isSynthetic check
2. wrote some unit test for DataSourcePropertiesConfiguration and SentinelDataSourceHandler

### Describe how to verify it
1. download jacoco library files:https://www.eclemma.org/jacoco/
2. run unit test with jacocoagent.jar, VM options like:
-javaagent:D:\Users\Test\Desktop\jacoco-0.8.7\lib/jacocoagent.jar=includes=*,output=tcpserver,port=6271,address=127.0.0.1,append=true

### CODE REVIEW

**Comment by hnyyghk** [comment]
邮件已收到，谢谢！------------------葛弘康QQ：824080252TEL：15927146347华中科技大学电信学院电子信息工程1405班

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/test/java/com/alibaba/cloud/sentinel/custom/SentinelDataSourceHandlerTests.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	@Before
+	public void setUp() {
+		beanFactory = mock(DefaultListableBeanFactory.class);
+		sentinelProperties = mock(SentinelProperties.class);
+		env = mock(Environment.class);
+		sentinelDataSourceHandler = new SentinelDataSourceHandler(beanFactory, sentinelProperties, env);
+	}
+
+	@Test
+	public void testParseBeanDefinition() {
```
Comment (DanielLiu1123) [inline spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/test/java/com/alibaba/cloud/sentinel/custom/SentinelDataSourceHandlerTests.java:65]: Add comments about test env.

File: spring-cloud-alibaba-starters/spring-cloud-alibaba-sentinel-datasource/src/test/java/com/alibaba/cloud/sentinel/datasource/DataSourcePropertiesConfigurationTests.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		apollo.setFlowRulesKey("test-flow-rules");
+		apollo.setDefaultFlowRuleValue("[]");
+		apollo.setDataType("json");
+		apollo.setRuleType(RuleType.FLOW);
+		configuration.setApollo(apollo);
+
+		List<String> validField = configuration.getValidField();
+
+		assertThat(validField.size()).isEqualTo(1);
+		assertThat(validField).doesNotContain("$jacocoData");
```
Comment (DanielLiu1123) [inline spring-cloud-alibaba-starters/spring-cloud-alibaba-sentinel-datasource/src/test/java/com/alibaba/cloud/sentinel/datasource/DataSourcePropertiesConfigurationTests.java:55]: Add some comment on this.

**Comment by DanielLiu1123** [comment]
You have code style problems. Please execute `mvn clean compile` and fix it.

**Comment by hnyyghk** [comment]
code style problems is fixed

**Pull request merged**